keys.go

```diff
@@ -548,6 +555,45 @@ func (c *Client) wrap(ctx context.Context, idOrAlias string, plainText []byte, a
 	return pt, ct, nil
 }
 
+// WrapWithKeyVersion function supports KeyVersion Details, PlainText and Cyphertext in response
+func (c *Client) WrapV2(ctx context.Context, idOrAlias string, plainText []byte, additionalAuthData *[]string) (*KeyActionResponse, error) {
+	keysActionReq := &KeysActionRequest{}
+	keyActionRes := &KeyActionResponse{}
+
+	if plainText != nil {
+		_, err := base64.StdEncoding.DecodeString(string(plainText))
+		if err != nil {
+			return keyActionRes, err
+		}
+		keysActionReq.PlainText = string(plainText)
+	}
+
+	if additionalAuthData != nil {
+		keysActionReq.AAD = *additionalAuthData
+	}
+
+	keysAction, err := c.doKeysAction(ctx, idOrAlias, "wrap", keysActionReq)
+	if err != nil {
+		return keyActionRes, err
+	}
+
+	keyActionRes = &KeyActionResponse{
+		PlainText:  keysAction.PlainText,
+		CipherText: keysAction.CipherText,
+	}
+	if keysAction.KeyVersion != nil {
+		keyActionRes.KeyVersion = &KeyVersion{
+			ID: keysAction.KeyVersion.ID,
+		}
+	}
+	if keysAction.KeyVersion.CreationDate != nil {
+		keyActionRes.KeyVersion = &KeyVersion{
+			CreationDate: keysAction.KeyVersion.CreationDate,
+		}
+	}
```

Comment:
We should be calling `keysAction.KeyVersion.CreationDate` after checking that `keysAction.KeyVersion != nil`. 
Also, in current implementation you would only have CreaationDate (and not ID)  in keyActionRes.KeyVersion struct as you are assigning a new `KeyVersion` struct again. 
```suggestion
	if keysAction.KeyVersion != nil {
		keyActionRes.KeyVersion = &KeyVersion{
			ID: keysAction.KeyVersion.ID,
		}
		if keysAction.KeyVersion.CreationDate != nil {
			keyActionRes.KeyVersion.CreationDate = keysAction.KeyVersion.CreationDate
		}
	}
```